geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/StartLocatorCommand.java

```diff
@@ -120,25 +121,32 @@ public ResultModel startLocator(
           help = CliStrings.START_LOCATOR__HTTP_SERVICE_BIND_ADDRESS__HELP) final String httpServiceBindAddress,
       @CliOption(key = CliStrings.START_LOCATOR__REDIRECT_OUTPUT, unspecifiedDefaultValue = "false",
           specifiedDefaultValue = "true",
-          help = CliStrings.START_LOCATOR__REDIRECT_OUTPUT__HELP) final Boolean redirectOutput)
+          help = CliStrings.START_LOCATOR__REDIRECT_OUTPUT__HELP) final Boolean redirectOutput,
+      @CliOption(key = CliStrings.START_LOCATOR__MEMBERSHIP_BIND_ADDRESS,
+          help = CliStrings.START_LOCATOR__MEMBERSHIP_BIND_ADDRESS__HELP) final String membershipBindAddress)
       throws Exception {
     if (StringUtils.isBlank(memberName)) {
       // when the user doesn't give us a name, we make one up!
       memberName = StartMemberUtils.getNameGenerator().generate('-');
     }
 
-    workingDirectory = StartMemberUtils.resolveWorkingDir(
-        workingDirectory == null ? null : new File(workingDirectory), new File(memberName));
+    workingDirectory = resolveWorkingDirectory(workingDirectory, memberName);
 
     return doStartLocator(memberName, bindAddress, classpath, force, group, hostnameForClients,
         jmxManagerHostnameForClients, includeSystemClasspath, locators, logLevel, mcastBindAddress,
         mcastPort, port, workingDirectory, gemfirePropertiesFile, gemfireSecurityPropertiesFile,
         initialHeap, maxHeap, jvmArgsOpts, connect, enableSharedConfiguration,
         loadSharedConfigurationFromDirectory, clusterConfigDir, httpServicePort,
-        httpServiceBindAddress, redirectOutput);
+        httpServiceBindAddress, redirectOutput, membershipBindAddress);
 
   }
 
+  @VisibleForTesting
+  protected static String resolveWorkingDirectory(String workDirValue, String memberName) {
```

Comment:
Do you really need the both `StartLocatorCommand.resolveWorkingDirectory(String, String)` and `StartMemberUtils.resolveWorkingDir`? I think you should either remove the new method or move it down to `StartMemberUtils`.

Reply:
`StartMemberUtils` is internal, so feel free to change method signatures in there or rename `resolveWorkingDir` to `resolveWorkingDirectory`. It's generally always best to spell out words rather than use abbreviations.

While you're in this code, I would also change `StartMemberUtils` and its constant `GEODE_HOME` to package-private. The only uses of this class are from within the same package.

Reply:
> I've marked some changes involving logging, AssertJ, and minor clean code design issues.
 
Thanks for the review @kirklund , I think I have fixed all the issues in the last commit, please take a look.